test/storage/checkpoint_test.cpp

```diff
@@ -0,0 +1,194 @@
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+#include "common/object_pool.h"
+#include "storage/checkpoint_manager.h"
+#include "storage/sql_table.h"
+#include "storage/storage_util.h"
+#include "util/random_test_util.h"
+#include "util/storage_test_util.h"
+#include "util/transaction_test_util.h"
+
+#define CHECKPOINT_FILE_PREFIX "checkpoint_"
+
+namespace terrier {
+
+struct CheckpointTests : public TerrierTest {};
+
+// NOLINTNEXTLINE
+TEST_F(CheckpointTests, SimpleCheckpointNoVarlen) {
+  const uint32_t num_rows = 100;
+  const uint32_t num_columns = 3;
```

Comment:
You need a significantly larger number of tests.  Specifically, you should test both 1000s of columns and 1000s of rows because your current test fits completely inside one 4 kb "page".  @yash620 identified that you are probably not resetting your page offset between "blocks", but your tests miss this because you never advance to a new block/page.